vcpp-client-reference/Source.cpp

```diff
@@ -1,22 +1,15 @@
 #pragma once
 
-#include <windows.h>
-
-#include <random>
-#include <sstream>
-#include <future>
-#include <chrono>
-#include <thread>
+#include "optimizer.grpc.pb.h"
 
 #include <google/protobuf/text_format.h>
 #include <grpcpp/grpcpp.h>
 
-// TODO: bad practice to reach to other directories? Maybe have build system make a local copy?
-#include "gen-src/optimizer.grpc.pb.h"
-#include "gen-src/optimizer.pb.h"
-
-//winsock2 needed by grpc
-#pragma comment(lib, "Ws2_32.lib")
+#include <chrono>
+#include <future>
+#include <random>
+#include <sstream>
+#include <thread>
```

Comment:
any reason for the order change?

Reply:
Most C++ coding conventions recommend putting local header files first. This typically means the include order is as follows:
1. Header file that corresponds to the current source file (e.g. `Source.hpp` if there is one)
2. Other project-local header files (e.g. `optimizer.grpc.pb.h` and headers in other directories such as `volition/api/foo.hpp`)
3. Third-party header files (e.g. `google/...`)
4. Standard library header files